internal/kubernetes/backup/backup.go

```diff
@@ -69,6 +70,29 @@ func FetchClusterBackup(ctx context.Context, opts ClusterBackupOptions) (*Cluste
 		}
 	}
 
+	// check that the cert-manager API versions are supported
+	crdClient, err := clients.NewCRDClient(opts.RestConfig)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create CRD client: %s", err)
+	}
+	var crds apiextensionsv1.CustomResourceDefinitionList
+	err = crdClient.List(ctx, &clients.GenericRequestOptions{}, &crds)
+	if err != nil {
+		return nil, fmt.Errorf("failed to list CRDs to determine the cert-manager API version in use: %s", err)
+	}
+	for _, crd := range crds.Items {
+		if crd.Spec.Group != "cert-manager.io" {
+			continue
+		}
+		if len(crd.Spec.Versions) == 0 {
+			return nil, fmt.Errorf("unexpectedly found no versions on cert-manager.io CRD %s", crd.Name)
+		}
+		// the first version in the list must be v1 as we expect v1 resources
+		if crd.Spec.Versions[0].Name != "v1" {
```

Comment:
Sorry @charlieegan3 I wasn't very clear on this. 
An unsupported cert-manager.io CRD would be one that does not have `v1` version. It is fine if they have `v1` *as well as* some of the alpha and beta versions- the CRDs between cert-manager v1 - v1.6 had all these versions, see i.e [this CRD](https://github.com/cert-manager/cert-manager/releases/download/v1.4.1/cert-manager.crds.yaml). It's not a problem for the backup command because as long as there is a v1 version, there will be version conversion happening and if we get custom resources at v1, anything that user created that is i.e `v1alpha2` will also be returned, so the backup will not miss any resources.

Let me know if this makes sense. Basically this check should just ensure that `v1` is one of the versions there-it's ok if other versions are also present.

Reply:
I was also wondering about this. I have done a little looking and I think you're right, but that v1 would also need to have `served: true` set. I'll update this now.

Reply:
I think that 7419e678b47e0a2dbb137700c3f5b23294d3fff1 addresses this now 🤞

Reply:
Thanks, I've tested with cert-manager 0.16 that only has alpha and beta versions and can seee that it works.